template/web/wp-content/themes/<%= slug %>/app/Managers/TwigManager.php

```diff
@@ -29,4 +33,48 @@ public function add_twig_template_from_string( \Twig\Environment $twig ) {
 		$twig->addExtension( new \Twig\Extension\StringLoaderExtension() );
 		return $twig;
 	}
+
+	/**
+	 * Adds template_from_string to Twig.
+	 *
+	 * @link https://github.com/djboris88/twig-commented-include
+	 * @param \Twig\Environment $twig The Twig environment.
+	 * @return \Twig\Environment
+	 */
+	public function add_twig_template_include_comments( \Twig\Environment $twig ) {
+		if ( getenv( 'APP_DEBUG' ) === 'false' ) {
+			return $twig;
+		}
+
+		$twig->addExtension( new \Djboris88\Twig\Extension\CommentedIncludeExtension() );
+		return $twig;
+	}
+
+	/**
+	 * Add debug comments to Timber::render
+	 *
+	 * @param string $output HTML.
+	 * @param array  $data data.
+	 * @param string $file name.
+	 * @return string
+	 */
+	public function add_twig_template_render_comments( $output, $data, $file ) {
+		if ( getenv( 'APP_DEBUG' ) === 'false' ) {
+			return $output;
+		}
+
+		return "\n<!-- Begin output of '" . $file . "' -->\n" . $output . "\n<!-- / End output of '" . $file . "' -->\n";
```

Comment:
Do we need both `add_twig_template_include_comments` and `add_twig_template_render_comments`? 🤔

Reply:
While it's close in behavior, i think this adding usefull information: 

`add_twig_template_include_comments` is adding comments for every included/embed files. 
`add_twig_template_render_comments` is adding one comment after each `Timber::render`, so you will get the comment `<!-- Begin output of pages/single.twig -->` which is super usefull (espacially if it's a specific template)